.github/workflows/test.yml

```
@@ -0,0 +1,45 @@
+name: test
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
+
+jobs:
+  test:
+    permissions:
+      contents: read      # for actions/checkout to fetch code
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        # test against the "oldest" supported version and the current version
+        # of go. Go 1.17 is kept in this matrix as it is the minimum version
+        # specified in go.mod, and maintaining compatibility with go 1.17 is
+        # currently not much of a burden. Most projects using this module are
+        # using newer versions than that, so we can drop the old version if
+        # it becomes too much of a burden.
+        go-version: [1.17.x, 1.19.x]
```

```suggestion
        go-version: [1.17.x, 'stable']
```

This lets us always default to the latest release: https://github.com/actions/setup-go#using-stableoldstable-aliases

Given the extremely low activity on this repo, I'd much rather use `stable` than having to bump the pins, because we're more likely to catch issues that way...

---

Interestingly, they also support picking up the version from `go.mod`, but doesn't like like it's possibly to cleanly do this as part of a matrix: https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file